### ISSUE TITLE
[Core] Ray actor reuses workers that has been previously running a task

### ISSUE THREAD

**Issue by scv119** [ray-project/ray#28773]
### What happened + What you expected to happen

his behavior looks like a bug.  at least our expectation is we always create a new worker for an actor https://docs.ray.io/en/latest/ray-core/actors.html

### Versions / Dependencies

latest master

### Reproduction script

```
@ray.remote
def f():
    print("300")

@ray.remote
class X:
    def f(self):
        print("actor X")
ray.get(f.remote())
y = X.remote()
ray.get(y.f.remote())

```

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by scv119** [comment]
huh looks this has been fixed in 2.0.0

**Comment by scv119** [comment]
https://github.com/ray-project/ray/pull/22482 fixed the issue.
